src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/MulticlassLogisticRegression.cs

```diff
@@ -993,35 +951,280 @@ DataViewRow ICanGetSummaryAsIRow.GetSummaryIRowOrNull(RoleMappedSchema schema)
+            {
+                Host.Assert(namedValues.Value is float);
+                writer.WriteLine("\t{0}\t{1}", namedValues.Key, (float)namedValues.Value);
+            }
+
+            if (Statistics != null)
+                Statistics.SaveText(writer, null, schema.Feature.Value, 20);
+        }
+    }
+
+    /// <summary>
+    /// Linear maximum entropy model of multiclass classifiers. It outputs classes probabilities.
+    /// This model is also known as multinomial logistic regression.
+    /// Please see https://en.wikipedia.org/wiki/Multinomial_logistic_regression for details.
+    /// </summary>
+    public sealed class MaximumEntropyModelParameters : LinearMulticlassModelParametersBase
+    {
+        internal const string LoaderSignature = "MultiClassLRExec";
+        internal const string RegistrationName = "MulticlassLogisticRegressionPredictor";
+
+        private static VersionInfo VersionInfo =>
+            new VersionInfo(
+                modelSignature: "MULTI LR",
+                // verWrittenCur: 0x00010001, // Initial
+                // verWrittenCur: 0x00010002, // Added class names
+                verWrittenCur: 0x00010003, // Added model stats
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(MaximumEntropyModelParameters).Assembly.FullName);
+
+        /// <summary>
+        /// Function used to pass <see cref="VersionInfo"/> into parent class. It may be used when saving the model.
+        /// </summary>
+        private protected override VersionInfo GetVersionInfo() => VersionInfo;
+
+        internal MaximumEntropyModelParameters(IHostEnvironment env, in VBuffer<float> weights, int numClasses, int numFeatures, string[] labelNames, LinearModelStatistics stats = null)
+            : base(env, RegistrationName, weights, numClasses, numFeatures, labelNames, stats)
+        {
+        }
+
+        internal MaximumEntropyModelParameters(IHostEnvironment env, VBuffer<float>[] weights, float[] bias, int numClasses, int numFeatures, string[] labelNames, LinearModelStatistics stats = null)
+            : base(env, RegistrationName, weights, bias, numClasses, numFeatures, labelNames, stats)
+        {
+        }
+
+        private MaximumEntropyModelParameters(IHostEnvironment env, ModelLoadContext ctx)
+            : base(env, RegistrationName, ctx)
+        {
+        }
+
+        /// <summary>
+        /// This function applies softmax to <paramref name="dst"/>. For details about softmax, see https://en.wikipedia.org/wiki/Softmax_function.
+        /// </summary>
+        /// <param name="dst">Score vector should be calibrated.</param>
+        private protected override void Calibrate(Span<float> dst)
+        {
+            Host.Assert(dst.Length >= NumberOfClasses);
+
```

Comment:
Why is `dst.Lenght> NumberOfClasses` is ok?

Reply:
Fixed.

---
In reply to: [266662595](https://github.com/dotnet/machinelearning/pull/2976#discussion_r266662595) [](ancestors = 266662595)